src/main/java/org/opensearch/security/privileges/ActionPrivileges.java

```diff
@@ -449,7 +456,58 @@ PrivilegesEvaluatorResponse providesPrivilege(PrivilegesEvaluationContext contex
                 }
             }
 
-            return PrivilegesEvaluatorResponse.insufficient(action);
```

Comment:
wdyt about creating 2 `providesPrivilege` methods. One for roles-based and one for permissions-based? The last arg of this method takes a set of roles so it already looks like its specifically designed for roll-based permission evaluation. Maybe there can be a branch in https://github.com/opensearch-project/security/blob/a6859519806bd9f22d6e653ff02aa3a5fc86e3de/src/main/java/org/opensearch/security/privileges/ActionPrivileges.java#L147-L149 depending on the type of PrivilegeEvaluationContext?

Reply:
Done in: [77ad6c3](https://github.com/opensearch-project/security/pull/4992/commits/77ad6c3722479750b86afd2984460e063defaf3b)